drivers/volume/portworx/portworx.go

```diff
@@ -2094,20 +2094,38 @@ func (d *portworx) EstimatePoolExpandSize(apRule apapi.AutopilotRule, pool node.
 					Operation: "Pool Condition Expression Key",
 				}
 			}
-
 			if doesConditionMatch(metricValue, conditionExpression) {
 				for _, ruleAction := range apRule.Spec.Actions {
-					actionScalePercentage, err := strconv.ParseUint(ruleAction.Params[aututils.RuleActionsScalePercentage], 10, 64)
-					if err != nil {
-						return 0, err
-					}
+					if actionScalePercentageValue, ok := ruleAction.Params[aututils.RuleActionsScalePercentage]; ok {
+						actionScalePercentage, err := strconv.ParseUint(actionScalePercentageValue, 10, 64)
+						if err != nil {
+							return 0, err
+						}
 
-					requiredScaleSize := float64(calculatedTotalSize * actionScalePercentage / 100)
-					if actionScaleType == aututils.RuleScaleTypeAddDisk {
-						requiredNewDisks := uint64(math.Ceil(requiredScaleSize / float64(baseDiskSize)))
-						calculatedTotalSize += requiredNewDisks * baseDiskSize
-					} else {
-						calculatedTotalSize += uint64(requiredScaleSize)
+						requiredScaleSize := float64(calculatedTotalSize * actionScalePercentage / 100)
+						if actionScaleType == aututils.RuleScaleTypeAddDisk {
+							requiredNewDisks := uint64(math.Ceil(requiredScaleSize / float64(baseDiskSize)))
+							calculatedTotalSize += requiredNewDisks * baseDiskSize
+						} else {
+							calculatedTotalSize += uint64(requiredScaleSize)
+						}
```

Comment:
these lines look like they are common with the same logic in else if. Can you check if we can move this common logic outside. So just calculate `requiredScaleSize` in the if and else if. And once we have that, move the logic to calculate calculatedTotalSize outside.

Reply:
fixed